### ISSUE TITLE
Suggestion: OneUI design library

### ISSUE THREAD

**Issue by Yanndroid** [tribalfs/GalaxyMaxHzPub#60]
I came across your app by chance and noticed that you're using Samsung's OneUI design, so I might have something that interests you. A friend and I are maintaining multiple libraries for bringing the OneUI design to third party apps. We made them by decompiling Samsung apps and now they completely replace the stock androidx libraries. In addition we have a library that contains some useful custom views based on the other libraries to make it easier to implement them.
If you're interested you can check them out here: https://github.com/OneUIProject

**Comment by tribalfs** [comment]
Thanks. 
That's a really great library.
Wish we have known this earlier.
Could have saved us some development time.

**Comment by tribalfs** [comment]
I am trying to use it but I am having errors implementing Dagger Hilt on it.

**Comment by tribalfs** [comment]
Having errors on  my activities after adding Hilt dependencies. Can't figure out how to resolve it.
`Cannot access 'androidx.core.app.OnMultiWindowModeChangedProvider' which is a supertype of 'com.tribalfs.oneuitest.SettingsActivity'. Check your module classpath for missing or conflicting dependencies
Cannot access 'androidx.core.app.OnNewIntentProvider' which is a supertype of 'com.tribalfs.oneuitest.SettingsActivity'. Check your module classpath for missing or conflicting dependencies
Cannot access 'androidx.core.app.OnPictureInPictureModeChangedProvider' which is a supertype of 'com.tribalfs.oneuitest.SettingsActivity'. Check your module classpath for missing or conflicting dependencies
Cannot access 'androidx.core.content.OnConfigurationChangedProvider' which is a supertype of 'com.tribalfs.oneuitest.SettingsActivity'. Check your module classpath for missing or conflicting dependencies
Cannot access 'androidx.core.content.OnTrimMemoryProvider' which is a supertype of 'com.tribalfs.oneuitest.SettingsActivity'. Check your module classpath for missing or conflicting dependencies
Cannot access 'androidx.core.view.MenuHost' which is a supertype of 'com.tribalfs.oneuitest.SettingsActivity'. Check your module classpath for missing or conflicting dependencies
`


Here's my test [repository](https://github.com/tribalfs/OneUiTest). The HiltImplementation branch is where I am having the error.

Thank you.

**Comment by tribalfs** [comment]
Got it working now. Lowering dagger version to [2.42](https://github.com/google/dagger/releases#:~:text=Compare-,Dagger%202.42,-What%E2%80%99s%20new%20in) removes the error.   Thank you.

**Comment by Yanndroid** [comment]
Yeah Samsung is a bit behind concerning their lib versions. So this can happen sometimes. Had a similar issue with kotlin where a lib was "too new". Glad you found the solution.

**Comment by tribalfs** [comment]
Take a look at [good guardians](https://www.apkmirror.com/apk/samsung-electronics-co-ltd/galaxy-labs/galaxy-labs-4-0-04-release/samsung-good-guardians-the-new-name-of-galaxy-labs-4-0-04-android-apk-download/download/?key=79b5a381850cef5d8dd909927f43b18bfed15936) app - it has non-R8 sesl5 modules
